{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Concatenating NCBI tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 946,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from os import path\n",
    "from shutil import copy\n",
    "from glob import glob\n",
    "import abacat as ab\n",
    "pd.set_option('display.max_rows', 200)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 947,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Reading\n",
    "syn = pd.read_csv(\"../data/ncbi/synechoccus_ncbi_metadata_27-05_dedup.csv\")\n",
    "cya = pd.read_csv(\"../data/ncbi/cyanos_gtdb_ncbi_metadata_07-05_dedup.csv\")\n",
    "\n",
    "# Removing intersection of rows from cya\n",
    "cya = cya[~cya['assembly_accession'].isin(syn['assembly_accession'])]\n",
    "\n",
    "# Adding columns to identify data sets: Synechococcus or GTDB cyanobacteria\n",
    "syn['is_synecho'] = True\n",
    "cya['is_synecho'] = False\n",
    "\n",
    "# Concatenate, format and save\n",
    "df = pd.concat((syn, cya))\n",
    "df['local_fna'] = df['filename'].apply(\n",
    "    lambda s: f\"../data/ncbi/processing/genomic_fna/{s}_genomic.fna\")\n",
    "df.to_csv(\"../data/ncbi/concatenated_ncbi_table.csv\")\n",
    "\n",
    "# Add aditional columns\n",
    "df['proteins_path'] = (\"../data/ncbi/processing/prodigal_proteins/\" +\n",
    "                    df['filename'] + \"_genomic_prodigal_proteins.faa\")\n",
    "df['genes_path'] = (\"../data/ncbi/processing/prodigal_genes/\" +\n",
    "                       df['filename'] + \"_genomic_prodigal_genes.fna\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 948,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We will remove GCA_003228185.1 due to being described as a mixed culture\n",
    "df = df[df['assembly_accession'] != 'GCA_003228185.1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 949,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create CheckM input batchfile\n",
    "copier = lambda x,y: [copy(i,y) for i in x]\n",
    "# copier(df[df[\"is_synecho\"]][\"local_fna\"], f\"../data/checkm/input/\")\n",
    "\n",
    "# Should try this:\n",
    "# map(lambda x,y: [copy(i,y) for i in x],\n",
    "#    (df[df[\"is_synecho\"]][\"local_fna\"], f\"../data/checkm/input/\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 950,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Join completeness and contamination\n",
    "\n",
    "# GTDB genomes\n",
    "gtdb_metadata = \"/home/vini/data/db/gtdb/bac120_metadata.tsv\"\n",
    "gtdb = pd.read_csv(gtdb_metadata, sep=\"\\t\")\n",
    "gtdb.rename(columns={\"ncbi_biosample\": \"biosample\"}, inplace=True)\n",
    "df = df.merge(gtdb[['biosample', 'checkm_completeness', 'checkm_contamination']],\n",
    "         on='biosample', how='left')\n",
    "\n",
    "# CheckM data\n",
    "checkm = pd.read_csv(\"../data/checkm/checkm_tax_wf.tsv\", sep=\"\\t\")\n",
    "checkm.rename(columns={\"Completeness\": \"checkm_completeness\",\n",
    "                       \"Contamination\": \"checkm_contamination\"}, inplace=True)\n",
    "checkm['filename'] = checkm[\"Bin Id\"].str.split(\"_genomic\", expand=True).iloc[:, 0]\n",
    "df = df.merge(checkm[['filename', 'checkm_completeness', 'checkm_contamination']],\n",
    "              on='filename', how='left')\n",
    "df['checkm_completeness'] = df['checkm_completeness_y'].fillna(\n",
    "    df['checkm_completeness_x'])\n",
    "df['checkm_contamination'] = df['checkm_contamination_y'].fillna(\n",
    "    df['checkm_contamination_x'])\n",
    "df = df.drop(['checkm_completeness_x', 'checkm_completeness_y',\n",
    "              'checkm_contamination_x', 'checkm_contamination_y'], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 951,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter genomes with < 50% completeness and > 5% contamination\n",
    "df['quality_filter'] = df.apply(lambda row: True if\n",
    "         row['checkm_completeness'] > 50 and\n",
    "         row['checkm_contamination'] < 10 else\n",
    "        False, axis=1)\n",
    "\n",
    "# Create GTDB batchfile for Synechococcus genomes\n",
    "# df[df['is_synecho']][['local_fna', 'assembly_accession']].to_csv(\n",
    "#     '../data/gtdb/batchfile.tsv', sep='\\t', index=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 952,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Join GTDB taxonomy for GTDB cyanobacteria\n",
    "gtdb_tax = gtdb[\"gtdb_taxonomy\"].str.split(\";\", expand=True)\n",
    "gtdb_tax.columns = (\"gtdb_\" + i for i in \"domain phylum class order family genus species\".split())\n",
    "gtdb_tax = gtdb_tax.applymap(lambda s: s[3:] if s is not None else s)\n",
    "gtdb[gtdb_tax.columns] = gtdb_tax\n",
    "df = df.merge(gtdb[['biosample'] + list(gtdb_tax.columns)], on='biosample', how='left')\n",
    "\n",
    "# Join GTDB taxonomy for NCBI Synechococcus genomes\n",
    "synecho_gtdb = pd.read_csv(\"../data/gtdb/output/synechococcus.bac120.summary.tsv\", sep=\"\\t\")\n",
    "gtdb_tax = synecho_gtdb[\"classification\"].str.split(\";\", expand=True)\n",
    "gtdb_tax.columns = (\"gtdb_\" + i for i in \"domain phylum class order family genus species\".split())\n",
    "gtdb_tax = gtdb_tax.applymap(lambda s: s[3:] if isinstance(s, str) else s)\n",
    "synecho_gtdb[gtdb_tax.columns] = gtdb_tax\n",
    "\n",
    "df = df.merge(synecho_gtdb.rename(columns={\"user_genome\": \"assembly_accession\"}), on='assembly_accession', how='left')\n",
    "\n",
    "for i in gtdb_tax.columns:\n",
    "    df[i] = df[i + '_y'].fillna(df[i + '_x'])\n",
    "    df = df.drop([i + \"_x\", i + \"_y\"], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 953,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add Abacat to calculate GC and size\n",
    "df['abacat'] = df['local_fna'].apply(lambda s: ab.SequenceFile(s, 'contigs'))\n",
    "df['gc'] = df['abacat'].apply(lambda seq: seq.gc)\n",
    "df['megabases'] = df['abacat'].apply(lambda seq: seq.bps / 10e5)\n",
    "df = df[[i for i in df.columns if i != 'abacat']]\n",
    "df.to_csv(\"../data/misc/metadata_v0.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### \"Checkpoint\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 968,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add labels from existing file\n",
    "df = pd.read_csv(\"../data/misc/metadata_v0.csv\")\n",
    "labels = pd.read_csv(\"../data/misc/labels.tsv\", sep=\"\\t\")\n",
    "labels.columns = (\"filename\", \"label\")\n",
    "labels['filename'] = labels['filename'].str.split(\"_genomic\", expand=True).iloc[:, 0]\n",
    "df = df.merge(labels, on='filename', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 970,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fill missing labels with GTDB tax\n",
    "def join_gtdb_tax(row):     \n",
    "    return \"_\".join([f\"{i[5]}_\" + str(row[i]) for i in gtdb_tax.columns[2:]])\n",
    "\n",
    "df['label'] = df.apply(lambda row: join_gtdb_tax(row) if row['label'] != row['label'] else row['label'], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 971,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Format columns\n",
    "def fill_rank(row, rank='genus'):\n",
    "    ranks = {\n",
    "        \"genus\": 0,\n",
    "        \"species\": 1,\n",
    "        \"strain\": 2\n",
    "    }\n",
    "    label = \"\"\n",
    "    if row['is_synecho']:\n",
    "        if ranks[rank] < 2:\n",
    "            label = str(row['label'].split(\"_\")[ranks[rank]])\n",
    "        else:\n",
    "            label = \"_\".join(row['label'].split(\"_\")[2:])\n",
    "        if len(label) > 1:\n",
    "            pass\n",
    "        else:\n",
    "            if ranks[rank] < 2:\n",
    "                label = str(row[f'gtdb_{rank}'])\n",
    "            else:\n",
    "                if row['isolate'] != row['isolate']:\n",
    "                    label = str(row['infraspecific_name'])\n",
    "                else:\n",
    "                    label = str(row['isolate'])\n",
    "    else:\n",
    "        if ranks[rank] == 2:\n",
    "            label = str(row['infraspecific_name'])\n",
    "        else:\n",
    "            label = str(row[f'gtdb_{rank}'])\n",
    "    return label\n",
    "\n",
    "df['new_genus'] = df.apply(lambda row: fill_rank(row, rank='genus'), axis=1)\n",
    "df['new_species'] = df.apply(lambda row: fill_rank(row, rank='species'), axis=1)\n",
    "df['new_strain'] = df.apply(lambda row: fill_rank(row, rank='strain'), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 972,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add Limnothrix and Cyanobium as synechos\n",
    "cyanobium = [\n",
    "    \"GFB01\", # not SC\n",
    "    \"NIES-981\", # not SC\n",
    "    \"PCC7001\", # 2 not SC\n",
    "    \"8F6\",\n",
    "    \"UBA5018\", # not SC\n",
    "    \"CCALA_15\", # not SC\n",
    "    \"PCC_6307\", # not SC, Cyanobium gracile (type species)\n",
    "    \"BO_8801\",\n",
    "    \"CACIAM_14\" # not SC\n",
    "]\n",
    "\n",
    "regnicoccus = [\n",
    "    \"WH-5701\" # 3 not SC\n",
    "]\n",
    "\n",
    "inmanicoccus = [\n",
    "    \"RCC307\" # 2 not SC\n",
    "]\n",
    "\n",
    "limnothrix = [ \n",
    "    \"Limnotrichaceae\", # 2 not in synecho\n",
    "    \"Enugrolinea\",\n",
    "    \"Limnothrix\" ]\n",
    "\n",
    "# Remove Synechos with only one genome\n",
    "singletons = [\n",
    "    \"Leptovivax\",\n",
    "    \"Coccusdissimilis\"\n",
    "]\n",
    "\n",
    "# for genus in singletons:\n",
    "#     df.loc[df[df['label'].apply(lambda s: any([i in s for i in list_]))].index, \"is_synecho\"] = False\n",
    "\n",
    "for list_, genus in zip((cyanobium, regnicoccus, inmanicoccus, limnothrix),\n",
    "                        (\"Cyanobium\", \"Regnicoccus\", \"Inmanicoccus\", \"Limnothrix\")):\n",
    "    df.loc[df[df['label'].apply(lambda s: any([i in s for i in list_]))].index, \"new_genus\"] = genus\n",
    "    df.loc[df[df['label'].apply(lambda s: any([i in s for i in list_]))].index, \"is_synecho\"] = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 973,
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_labels = pd.read_csv(\"../data/misc/missing_labels.csv\")\n",
    "\n",
    "df = df.merge(missing_labels, on='assembly_accession', how='left')\n",
    "for i in missing_labels.columns[1:]:\n",
    "    df[i] = df[i + '_y'].fillna(df[i + '_x'])\n",
    "    df = df.drop([i + \"_x\", i + \"_y\"], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 974,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Final edits\n",
    "\n",
    "# Drop duplicates and other classes\n",
    "df = df.query(\"gtdb_class == 'Cyanobacteriia'\")\n",
    "df = df.drop_duplicates(\"assembly_accession\")\n",
    "\n",
    "df['new_genus'] = df['new_genus'].str.replace('Enugrolinea', 'Limnothrix')\n",
    "df['new_genus'] = df['new_genus'].str.replace('Synechococcus-C', 'Synechococcus_C')\n",
    "\n",
    "# Manual edit to Vulcanococcus limneticus\n",
    "ix = df[df['organism_name'].str.contains(\"limneticus\")].index\n",
    "df.loc[ix, 'new_species'] = 'limneticus'\n",
    "df.loc[ix, 'new_strain']  = 'LL'\n",
    "df.loc[ix, 'label'] = df.loc[ix, 'organism_name']\n",
    "df.loc[ix, 'is_synecho'] = True\n",
    "\n",
    "# And to nan labels:\n",
    "df['label'] = df.apply(lambda row: row['organism_name'] if 'nan' in row['label'] else row['label'], axis=1)\n",
    "df['label'] = df['label'].str.replace(\"Magnicoccus\", \"Vulcanococcus\")\n",
    "df['new_genus'] = df['new_genus'].str.replace(\"Magnicoccus\", \"Vulcanococcus\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 975,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"../data/misc/metadata_v1.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 976,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Create AAI input for Synechococcus - added Cyanobium and Limnothrix\n",
    "dst = \"../data/comparem/input/\"\n",
    "\n",
    "#df[(df['is_synecho'])]['proteins_path'].apply(lambda file: copy(file.replace(\"..//ncbi\", \"../data/ncbi/\"), dst))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 977,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"../data/misc/metadata_v1.csv\")\n",
    "\n",
    "# Creating GTT input\n",
    "df = df[df['quality_filter']].drop_duplicates(\"label\")\n",
    "df['protein_filename'] = df['proteins_path'].apply(lambda s: path.basename(s))\n",
    "df['proteins_path'] = df['proteins_path'].str.replace(\"../data\", \"../\")\n",
    "df['gtt_label'] = df['label'] + \" # \" + df['assembly_accession']\n",
    "\n",
    "remove_from_gtt = [\"PCC-7407\", \"Gloeomargarita\", \"kilaueensis\", \"CACIAM-69\", \"RSCC\", \"Gloeocapsa\", \"Prochloraceae\", \"Phormidiaceae_A\"]\n",
    "df = df[~df.index.isin(df[df['gtt_label'].apply(lambda s: any(i for i in remove_from_gtt if i in s))].index)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 979,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Exporting\n",
    "df['proteins_path'].to_csv(\"../data/gtt/1085_input.txt\", index=False, header=False)\n",
    "df[['protein_filename', 'gtt_label']].to_csv(\"../data/gtt/1085_labels.tsv\", sep=\"\\t\", index=False, header=False)\n",
    "df.to_csv(\"../data/metadata.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 898,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Building table 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 899,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(253, 61)"
      ]
     },
     "execution_count": 899,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['is_synecho']].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 893,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>biosample</th>\n",
       "      <th>wgs_master</th>\n",
       "      <th>excluded_from_refseq</th>\n",
       "      <th>refseq_category</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>taxid</th>\n",
       "      <th>species_taxid</th>\n",
       "      <th>isolate</th>\n",
       "      <th>...</th>\n",
       "      <th>gtdb_family</th>\n",
       "      <th>gtdb_genus</th>\n",
       "      <th>gtdb_species</th>\n",
       "      <th>gc</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>infraspecific_name</th>\n",
       "      <th>new_genus</th>\n",
       "      <th>new_species</th>\n",
       "      <th>new_strain</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>GCA_000155595.1</td>\n",
       "      <td>PRJNA19377</td>\n",
       "      <td>SAMN02436177</td>\n",
       "      <td>ABRV00000000.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>91464</td>\n",
       "      <td>91464</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Phormidesmiaceae</td>\n",
       "      <td>Phormidesmis</td>\n",
       "      <td>Phormidesmis sp000155595</td>\n",
       "      <td>0.4816</td>\n",
       "      <td>Synechococcus sp. PCC 7335</td>\n",
       "      <td>strain=PCC 7335</td>\n",
       "      <td>Coccusdissimilis</td>\n",
       "      <td>mexicanus</td>\n",
       "      <td>PCC_7335</td>\n",
       "      <td>Coccusdissimilis_mexicanus_PCC_7335</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   assembly_accession  bioproject     biosample      wgs_master  \\\n",
       "21    GCA_000155595.1  PRJNA19377  SAMN02436177  ABRV00000000.1   \n",
       "\n",
       "   excluded_from_refseq refseq_category relation_to_type_material  taxid  \\\n",
       "21                  NaN              na                       NaN  91464   \n",
       "\n",
       "    species_taxid isolate  ...       gtdb_family    gtdb_genus  \\\n",
       "21          91464     NaN  ...  Phormidesmiaceae  Phormidesmis   \n",
       "\n",
       "                gtdb_species      gc               organism_name  \\\n",
       "21  Phormidesmis sp000155595  0.4816  Synechococcus sp. PCC 7335   \n",
       "\n",
       "   infraspecific_name         new_genus new_species new_strain  \\\n",
       "21    strain=PCC 7335  Coccusdissimilis   mexicanus   PCC_7335   \n",
       "\n",
       "                                  label  \n",
       "21  Coccusdissimilis_mexicanus_PCC_7335  \n",
       "\n",
       "[1 rows x 61 columns]"
      ]
     },
     "execution_count": 893,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['label'].str.contains(\"mexicanus\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 833,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Synechococcales          167\n",
       "Cyanobacteriales          14\n",
       "PCC-7336                  11\n",
       "Thermosynechococcales      7\n",
       "Neosynechococcales         1\n",
       "Phormidesmiales            1\n",
       "Pseudanabaenales           1\n",
       "Name: gtdb_order, dtype: int64"
      ]
     },
     "execution_count": 833,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['is_synecho']]['gtdb_order'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 834,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Cyanobiaceae              161\n",
       "Limnotrichaceae            14\n",
       "JA-3-3Ab                   10\n",
       "Thermosynechococcaceae      7\n",
       "Synechococcaceae            6\n",
       "Phormidesmiaceae            1\n",
       "Neosynechococcaceae         1\n",
       "PCC-7336                    1\n",
       "Pseudanabaenaceae           1\n",
       "Name: gtdb_family, dtype: int64"
      ]
     },
     "execution_count": 834,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['is_synecho']]['gtdb_family'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 835,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Limnothrix             14\n",
       "Leptococcus            10\n",
       "Thermosynechococcus     6\n",
       "Synechococcus           6\n",
       "Brevicoccus             1\n",
       "Leptovivax              1\n",
       "Stenotopis              1\n",
       "Coccusdissimilis        1\n",
       "Neosynechococcus        1\n",
       "Name: new_genus, dtype: int64"
      ]
     },
     "execution_count": 835,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['is_synecho'] & (df['gtdb_family'] != 'Cyanobiaceae') & df['quality_filter']]['new_genus'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 836,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>biosample</th>\n",
       "      <th>wgs_master</th>\n",
       "      <th>excluded_from_refseq</th>\n",
       "      <th>refseq_category</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>taxid</th>\n",
       "      <th>species_taxid</th>\n",
       "      <th>isolate</th>\n",
       "      <th>...</th>\n",
       "      <th>gtdb_species</th>\n",
       "      <th>gc</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>infraspecific_name</th>\n",
       "      <th>new_genus</th>\n",
       "      <th>new_species</th>\n",
       "      <th>new_strain</th>\n",
       "      <th>label</th>\n",
       "      <th>protein_filename</th>\n",
       "      <th>gtt_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>GCA_000010065.1</td>\n",
       "      <td>PRJNA13282</td>\n",
       "      <td>SAMD00061073</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>269084</td>\n",
       "      <td>32046</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Synechococcus elongatus</td>\n",
       "      <td>0.55484</td>\n",
       "      <td>Synechococcus elongatus PCC 6301</td>\n",
       "      <td>strain=PCC 6301</td>\n",
       "      <td>Synechococcus</td>\n",
       "      <td>elongatus</td>\n",
       "      <td>PCC_6301</td>\n",
       "      <td>Synechococcus_elongatus_PCC_6301</td>\n",
       "      <td>GCA_000010065.1_ASM1006v1_genomic_prodigal_pro...</td>\n",
       "      <td>Synechococcus_elongatus_PCC_6301 # GCA_0000100...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>GCA_000012525.1</td>\n",
       "      <td>PRJNA10645</td>\n",
       "      <td>SAMN02598254</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1140</td>\n",
       "      <td>32046</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Synechococcus elongatus</td>\n",
       "      <td>0.55426</td>\n",
       "      <td>Synechococcus elongatus PCC 7942 = FACHB-805</td>\n",
       "      <td>strain=PCC 7942</td>\n",
       "      <td>Synechococcus</td>\n",
       "      <td>elongatus</td>\n",
       "      <td>PCC_7942</td>\n",
       "      <td>Synechococcus_elongatus_PCC_7942</td>\n",
       "      <td>GCA_000012525.1_ASM1252v1_genomic_prodigal_pro...</td>\n",
       "      <td>Synechococcus_elongatus_PCC_7942 # GCA_0000125...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>GCA_003846445.1</td>\n",
       "      <td>PRJNA476044</td>\n",
       "      <td>SAMN09425400</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2219813</td>\n",
       "      <td>32046</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.54900</td>\n",
       "      <td>Synechococcus elongatus PCC 11801</td>\n",
       "      <td>PCC 11801</td>\n",
       "      <td>Synechococcus</td>\n",
       "      <td>elongatus</td>\n",
       "      <td>PCC 11801</td>\n",
       "      <td>Synechococcus_sp._PCC 11801</td>\n",
       "      <td>GCA_003846445.1_ASM384644v1_genomic_prodigal_p...</td>\n",
       "      <td>Synechococcus_sp._PCC 11801 # GCA_003846445.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>GCA_003957805.1</td>\n",
       "      <td>PRJNA496333</td>\n",
       "      <td>SAMN09724271</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2479200</td>\n",
       "      <td>32046</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Synechococcus elongatus</td>\n",
       "      <td>0.55123</td>\n",
       "      <td>Synechococcus elongatus UTEX 3055</td>\n",
       "      <td>strain=UTEX 3055</td>\n",
       "      <td>Synechococcus</td>\n",
       "      <td>elongatus</td>\n",
       "      <td>UTEX_3055</td>\n",
       "      <td>Synechococcus_elongatus_UTEX_3055</td>\n",
       "      <td>GCA_003957805.1_ASM395780v1_genomic_prodigal_p...</td>\n",
       "      <td>Synechococcus_elongatus_UTEX_3055 # GCA_003957...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>GCA_009498715.1</td>\n",
       "      <td>PRJNA510697</td>\n",
       "      <td>SAMN10613228</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2283154</td>\n",
       "      <td>32046</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.54793</td>\n",
       "      <td>Synechococcus elongatus PCC 11802</td>\n",
       "      <td>PCC 11802</td>\n",
       "      <td>Synechococcus</td>\n",
       "      <td>elongatus</td>\n",
       "      <td>PCC 11802</td>\n",
       "      <td>Synechococcus_sp._PCC 11802</td>\n",
       "      <td>GCA_009498715.1_ASM949871v1_genomic_prodigal_p...</td>\n",
       "      <td>Synechococcus_sp._PCC 11802 # GCA_009498715.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>88</td>\n",
       "      <td>GCA_000817325.1</td>\n",
       "      <td>PRJNA209528</td>\n",
       "      <td>SAMN03278348</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1350461</td>\n",
       "      <td>1350461</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Synechococcus elongatus</td>\n",
       "      <td>0.55436</td>\n",
       "      <td>Synechococcus sp. UTEX 2973</td>\n",
       "      <td>strain=UTEX 2973</td>\n",
       "      <td>Synechococcus</td>\n",
       "      <td>elongatus</td>\n",
       "      <td>UTEX_2973</td>\n",
       "      <td>Synechococcus_elongatus_UTEX_2973</td>\n",
       "      <td>GCA_000817325.1_ASM81732v1_genomic_prodigal_pr...</td>\n",
       "      <td>Synechococcus_elongatus_UTEX_2973 # GCA_000817...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6 rows × 63 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   assembly_accession   bioproject     biosample wgs_master  \\\n",
       "6     GCA_000010065.1   PRJNA13282  SAMD00061073        NaN   \n",
       "7     GCA_000012525.1   PRJNA10645  SAMN02598254        NaN   \n",
       "8     GCA_003846445.1  PRJNA476044  SAMN09425400        NaN   \n",
       "9     GCA_003957805.1  PRJNA496333  SAMN09724271        NaN   \n",
       "10    GCA_009498715.1  PRJNA510697  SAMN10613228        NaN   \n",
       "88    GCA_000817325.1  PRJNA209528  SAMN03278348        NaN   \n",
       "\n",
       "   excluded_from_refseq        refseq_category    relation_to_type_material  \\\n",
       "6                   NaN  representative genome  assembly from type material   \n",
       "7                   NaN                     na                          NaN   \n",
       "8                   NaN                     na                          NaN   \n",
       "9                   NaN                     na                          NaN   \n",
       "10                  NaN                     na                          NaN   \n",
       "88                  NaN                     na                          NaN   \n",
       "\n",
       "      taxid  species_taxid isolate  ...             gtdb_species       gc  \\\n",
       "6    269084          32046     NaN  ...  Synechococcus elongatus  0.55484   \n",
       "7      1140          32046     NaN  ...  Synechococcus elongatus  0.55426   \n",
       "8   2219813          32046     NaN  ...                      NaN  0.54900   \n",
       "9   2479200          32046     NaN  ...  Synechococcus elongatus  0.55123   \n",
       "10  2283154          32046     NaN  ...                      NaN  0.54793   \n",
       "88  1350461        1350461     NaN  ...  Synechococcus elongatus  0.55436   \n",
       "\n",
       "                                   organism_name infraspecific_name  \\\n",
       "6               Synechococcus elongatus PCC 6301    strain=PCC 6301   \n",
       "7   Synechococcus elongatus PCC 7942 = FACHB-805    strain=PCC 7942   \n",
       "8              Synechococcus elongatus PCC 11801          PCC 11801   \n",
       "9              Synechococcus elongatus UTEX 3055   strain=UTEX 3055   \n",
       "10             Synechococcus elongatus PCC 11802          PCC 11802   \n",
       "88                   Synechococcus sp. UTEX 2973   strain=UTEX 2973   \n",
       "\n",
       "        new_genus new_species new_strain                              label  \\\n",
       "6   Synechococcus   elongatus   PCC_6301   Synechococcus_elongatus_PCC_6301   \n",
       "7   Synechococcus   elongatus   PCC_7942   Synechococcus_elongatus_PCC_7942   \n",
       "8   Synechococcus   elongatus  PCC 11801        Synechococcus_sp._PCC 11801   \n",
       "9   Synechococcus   elongatus  UTEX_3055  Synechococcus_elongatus_UTEX_3055   \n",
       "10  Synechococcus   elongatus  PCC 11802        Synechococcus_sp._PCC 11802   \n",
       "88  Synechococcus   elongatus  UTEX_2973  Synechococcus_elongatus_UTEX_2973   \n",
       "\n",
       "                                     protein_filename  \\\n",
       "6   GCA_000010065.1_ASM1006v1_genomic_prodigal_pro...   \n",
       "7   GCA_000012525.1_ASM1252v1_genomic_prodigal_pro...   \n",
       "8   GCA_003846445.1_ASM384644v1_genomic_prodigal_p...   \n",
       "9   GCA_003957805.1_ASM395780v1_genomic_prodigal_p...   \n",
       "10  GCA_009498715.1_ASM949871v1_genomic_prodigal_p...   \n",
       "88  GCA_000817325.1_ASM81732v1_genomic_prodigal_pr...   \n",
       "\n",
       "                                            gtt_label  \n",
       "6   Synechococcus_elongatus_PCC_6301 # GCA_0000100...  \n",
       "7   Synechococcus_elongatus_PCC_7942 # GCA_0000125...  \n",
       "8       Synechococcus_sp._PCC 11801 # GCA_003846445.1  \n",
       "9   Synechococcus_elongatus_UTEX_3055 # GCA_003957...  \n",
       "10      Synechococcus_sp._PCC 11802 # GCA_009498715.1  \n",
       "88  Synechococcus_elongatus_UTEX_2973 # GCA_000817...  \n",
       "\n",
       "[6 rows x 63 columns]"
      ]
     },
     "execution_count": 836,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['new_genus'] == 'Synechococcus']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 791,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Parasynechococcus      53\n",
       "Pseudosynechococcus    41\n",
       "Synechospongium        30\n",
       "Inmanicoccus           12\n",
       "Cyanobium               9\n",
       "Regnicoccus             9\n",
       "Vulcanococcus           4\n",
       "Lacustricoccus          3\n",
       "Name: new_genus, dtype: int64"
      ]
     },
     "execution_count": 791,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['is_synecho'] & (df['gtdb_family'] == 'Cyanobiaceae') & df['quality_filter']]['new_genus'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 837,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Synechococcales          167\n",
       "Cyanobacteriales          14\n",
       "PCC-7336                  11\n",
       "Thermosynechococcales      7\n",
       "Neosynechococcales         1\n",
       "Phormidesmiales            1\n",
       "Pseudanabaenales           1\n",
       "Name: gtdb_order, dtype: int64"
      ]
     },
     "execution_count": 837,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['is_synecho']]['gtdb_order'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 851,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "86    Pseudanabaenaceae\n",
       "Name: gtdb_family, dtype: object"
      ]
     },
     "execution_count": 851,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['is_synecho'] & (df['gtdb_order'] == 'Pseudanabaenales')]['gtdb_family']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 859,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>biosample</th>\n",
       "      <th>wgs_master</th>\n",
       "      <th>excluded_from_refseq</th>\n",
       "      <th>refseq_category</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>taxid</th>\n",
       "      <th>species_taxid</th>\n",
       "      <th>isolate</th>\n",
       "      <th>...</th>\n",
       "      <th>gtdb_species</th>\n",
       "      <th>gc</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>infraspecific_name</th>\n",
       "      <th>new_genus</th>\n",
       "      <th>new_species</th>\n",
       "      <th>new_strain</th>\n",
       "      <th>label</th>\n",
       "      <th>protein_filename</th>\n",
       "      <th>gtt_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>300</td>\n",
       "      <td>GCA_000007925.1</td>\n",
       "      <td>PRJNA419</td>\n",
       "      <td>SAMN02603142</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>167539</td>\n",
       "      <td>1219</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Prochlorococcus marinus</td>\n",
       "      <td>0.36442</td>\n",
       "      <td>Prochlorococcus marinus subsp. marinus str. CC...</td>\n",
       "      <td>strain=CCMP1375; SS120</td>\n",
       "      <td>Prochlorococcus</td>\n",
       "      <td>Prochlorococcus marinus</td>\n",
       "      <td>strain=CCMP1375; SS120</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "      <td>GCA_000007925.1_ASM792v1_genomic_prodigal_prot...</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>301</td>\n",
       "      <td>GCA_000011465.1</td>\n",
       "      <td>PRJNA213</td>\n",
       "      <td>SAMEA3138209</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>59919</td>\n",
       "      <td>1219</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Prochlorococcus_A marinus_S</td>\n",
       "      <td>0.30799</td>\n",
       "      <td>Prochlorococcus marinus subsp. pastoris str. C...</td>\n",
       "      <td>strain=MED4</td>\n",
       "      <td>Prochlorococcus_A</td>\n",
       "      <td>Prochlorococcus_A marinus_S</td>\n",
       "      <td>strain=MED4</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "      <td>GCA_000011465.1_ASM1146v1_genomic_prodigal_pro...</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>302</td>\n",
       "      <td>GCA_000011485.1</td>\n",
       "      <td>PRJNA220</td>\n",
       "      <td>SAMEA3138210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>74547</td>\n",
       "      <td>1219</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Prochlorococcus_C marinus_B</td>\n",
       "      <td>0.50740</td>\n",
       "      <td>Prochlorococcus marinus str. MIT 9313</td>\n",
       "      <td>strain=MIT9313</td>\n",
       "      <td>Prochlorococcus_C</td>\n",
       "      <td>Prochlorococcus_C marinus_B</td>\n",
       "      <td>strain=MIT9313</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "      <td>GCA_000011485.1_ASM1148v1_genomic_prodigal_pro...</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>303</td>\n",
       "      <td>GCA_000012465.1</td>\n",
       "      <td>PRJNA13911</td>\n",
       "      <td>SAMN00623057</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>59920</td>\n",
       "      <td>1219</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Prochlorococcus_B marinus_B</td>\n",
       "      <td>0.35123</td>\n",
       "      <td>Prochlorococcus marinus str. NATL2A</td>\n",
       "      <td>strain=NATL2A</td>\n",
       "      <td>Prochlorococcus_B</td>\n",
       "      <td>Prochlorococcus_B marinus_B</td>\n",
       "      <td>strain=NATL2A</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "      <td>GCA_000012465.1_ASM1246v1_genomic_prodigal_pro...</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>304</td>\n",
       "      <td>GCA_000012645.1</td>\n",
       "      <td>PRJNA13910</td>\n",
       "      <td>SAMN02598321</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>74546</td>\n",
       "      <td>1219</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Prochlorococcus_A marinus_L</td>\n",
       "      <td>0.31210</td>\n",
       "      <td>Prochlorococcus marinus str. MIT 9312</td>\n",
       "      <td>strain=MIT 9312</td>\n",
       "      <td>Prochlorococcus_A</td>\n",
       "      <td>Prochlorococcus_A marinus_L</td>\n",
       "      <td>strain=MIT 9312</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "      <td>GCA_000012645.1_ASM1264v1_genomic_prodigal_pro...</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1130</td>\n",
       "      <td>GCA_003211895.1</td>\n",
       "      <td>PRJNA445865</td>\n",
       "      <td>SAMN08886554</td>\n",
       "      <td>QCRJ00000000.1</td>\n",
       "      <td>derived from single cell</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2162537</td>\n",
       "      <td>2162537</td>\n",
       "      <td>AG-459-P07</td>\n",
       "      <td>...</td>\n",
       "      <td>Prochlorococcus_B marinus_B</td>\n",
       "      <td>0.35048</td>\n",
       "      <td>Prochlorococcus sp. AG-459-P07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Prochlorococcus_B</td>\n",
       "      <td>Prochlorococcus_B marinus_B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "      <td>GCA_003211895.1_ASM321189v1_genomic_prodigal_p...</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1131</td>\n",
       "      <td>GCA_003211315.1</td>\n",
       "      <td>PRJNA445865</td>\n",
       "      <td>SAMN08886599</td>\n",
       "      <td>QCTC00000000.1</td>\n",
       "      <td>derived from single cell</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2162543</td>\n",
       "      <td>2162543</td>\n",
       "      <td>AG-670-O17</td>\n",
       "      <td>...</td>\n",
       "      <td>Prochlorococcus_A sp003211315</td>\n",
       "      <td>0.30892</td>\n",
       "      <td>Prochlorococcus sp. AG-670-O17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Prochlorococcus_A</td>\n",
       "      <td>Prochlorococcus_A sp003211315</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "      <td>GCA_003211315.1_ASM321131v1_genomic_prodigal_p...</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1132</td>\n",
       "      <td>GCA_003211155.1</td>\n",
       "      <td>PRJNA445865</td>\n",
       "      <td>SAMN08886609</td>\n",
       "      <td>QCTM00000000.1</td>\n",
       "      <td>derived from single cell</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2162544</td>\n",
       "      <td>2162544</td>\n",
       "      <td>AG-673-K22</td>\n",
       "      <td>...</td>\n",
       "      <td>Prochlorococcus_B marinus_B</td>\n",
       "      <td>0.35286</td>\n",
       "      <td>Prochlorococcus sp. AG-673-K22</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Prochlorococcus_B</td>\n",
       "      <td>Prochlorococcus_B marinus_B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "      <td>GCA_003211155.1_ASM321115v1_genomic_prodigal_p...</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1133</td>\n",
       "      <td>GCA_003210915.1</td>\n",
       "      <td>PRJNA445865</td>\n",
       "      <td>SAMN08886623</td>\n",
       "      <td>QCUA00000000.1</td>\n",
       "      <td>derived from single cell</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2162546</td>\n",
       "      <td>2162546</td>\n",
       "      <td>AG-676-L21</td>\n",
       "      <td>...</td>\n",
       "      <td>Prochlorococcus_A sp003210915</td>\n",
       "      <td>0.30848</td>\n",
       "      <td>Prochlorococcus sp. AG-676-L21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Prochlorococcus_A</td>\n",
       "      <td>Prochlorococcus_A sp003210915</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "      <td>GCA_003210915.1_ASM321091v1_genomic_prodigal_p...</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1134</td>\n",
       "      <td>GCA_003208845.1</td>\n",
       "      <td>PRJNA445865</td>\n",
       "      <td>SAMN08886642</td>\n",
       "      <td>QCUT00000000.1</td>\n",
       "      <td>derived from single cell</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2162547</td>\n",
       "      <td>2162547</td>\n",
       "      <td>AG-679-N20</td>\n",
       "      <td>...</td>\n",
       "      <td>Prochlorococcus_A marinus_S</td>\n",
       "      <td>0.31032</td>\n",
       "      <td>Prochlorococcus sp. AG-679-N20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Prochlorococcus_A</td>\n",
       "      <td>Prochlorococcus_A marinus_S</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "      <td>GCA_003208845.1_ASM320884v1_genomic_prodigal_p...</td>\n",
       "      <td>c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>483 rows × 63 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     assembly_accession   bioproject     biosample      wgs_master  \\\n",
       "300     GCA_000007925.1     PRJNA419  SAMN02603142             NaN   \n",
       "301     GCA_000011465.1     PRJNA213  SAMEA3138209             NaN   \n",
       "302     GCA_000011485.1     PRJNA220  SAMEA3138210             NaN   \n",
       "303     GCA_000012465.1   PRJNA13911  SAMN00623057             NaN   \n",
       "304     GCA_000012645.1   PRJNA13910  SAMN02598321             NaN   \n",
       "...                 ...          ...           ...             ...   \n",
       "1130    GCA_003211895.1  PRJNA445865  SAMN08886554  QCRJ00000000.1   \n",
       "1131    GCA_003211315.1  PRJNA445865  SAMN08886599  QCTC00000000.1   \n",
       "1132    GCA_003211155.1  PRJNA445865  SAMN08886609  QCTM00000000.1   \n",
       "1133    GCA_003210915.1  PRJNA445865  SAMN08886623  QCUA00000000.1   \n",
       "1134    GCA_003208845.1  PRJNA445865  SAMN08886642  QCUT00000000.1   \n",
       "\n",
       "          excluded_from_refseq        refseq_category  \\\n",
       "300                        NaN  representative genome   \n",
       "301                        NaN                     na   \n",
       "302                        NaN                     na   \n",
       "303                        NaN                     na   \n",
       "304                        NaN                     na   \n",
       "...                        ...                    ...   \n",
       "1130  derived from single cell                     na   \n",
       "1131  derived from single cell                     na   \n",
       "1132  derived from single cell                     na   \n",
       "1133  derived from single cell                     na   \n",
       "1134  derived from single cell                     na   \n",
       "\n",
       "        relation_to_type_material    taxid  species_taxid     isolate  ...  \\\n",
       "300   assembly from type material   167539           1219         NaN  ...   \n",
       "301                           NaN    59919           1219         NaN  ...   \n",
       "302                           NaN    74547           1219         NaN  ...   \n",
       "303                           NaN    59920           1219         NaN  ...   \n",
       "304                           NaN    74546           1219         NaN  ...   \n",
       "...                           ...      ...            ...         ...  ...   \n",
       "1130                          NaN  2162537        2162537  AG-459-P07  ...   \n",
       "1131                          NaN  2162543        2162543  AG-670-O17  ...   \n",
       "1132                          NaN  2162544        2162544  AG-673-K22  ...   \n",
       "1133                          NaN  2162546        2162546  AG-676-L21  ...   \n",
       "1134                          NaN  2162547        2162547  AG-679-N20  ...   \n",
       "\n",
       "                       gtdb_species       gc  \\\n",
       "300         Prochlorococcus marinus  0.36442   \n",
       "301     Prochlorococcus_A marinus_S  0.30799   \n",
       "302     Prochlorococcus_C marinus_B  0.50740   \n",
       "303     Prochlorococcus_B marinus_B  0.35123   \n",
       "304     Prochlorococcus_A marinus_L  0.31210   \n",
       "...                             ...      ...   \n",
       "1130    Prochlorococcus_B marinus_B  0.35048   \n",
       "1131  Prochlorococcus_A sp003211315  0.30892   \n",
       "1132    Prochlorococcus_B marinus_B  0.35286   \n",
       "1133  Prochlorococcus_A sp003210915  0.30848   \n",
       "1134    Prochlorococcus_A marinus_S  0.31032   \n",
       "\n",
       "                                          organism_name  \\\n",
       "300   Prochlorococcus marinus subsp. marinus str. CC...   \n",
       "301   Prochlorococcus marinus subsp. pastoris str. C...   \n",
       "302               Prochlorococcus marinus str. MIT 9313   \n",
       "303                 Prochlorococcus marinus str. NATL2A   \n",
       "304               Prochlorococcus marinus str. MIT 9312   \n",
       "...                                                 ...   \n",
       "1130                     Prochlorococcus sp. AG-459-P07   \n",
       "1131                     Prochlorococcus sp. AG-670-O17   \n",
       "1132                     Prochlorococcus sp. AG-673-K22   \n",
       "1133                     Prochlorococcus sp. AG-676-L21   \n",
       "1134                     Prochlorococcus sp. AG-679-N20   \n",
       "\n",
       "          infraspecific_name          new_genus  \\\n",
       "300   strain=CCMP1375; SS120    Prochlorococcus   \n",
       "301              strain=MED4  Prochlorococcus_A   \n",
       "302           strain=MIT9313  Prochlorococcus_C   \n",
       "303            strain=NATL2A  Prochlorococcus_B   \n",
       "304          strain=MIT 9312  Prochlorococcus_A   \n",
       "...                      ...                ...   \n",
       "1130                     NaN  Prochlorococcus_B   \n",
       "1131                     NaN  Prochlorococcus_A   \n",
       "1132                     NaN  Prochlorococcus_B   \n",
       "1133                     NaN  Prochlorococcus_A   \n",
       "1134                     NaN  Prochlorococcus_A   \n",
       "\n",
       "                        new_species              new_strain  \\\n",
       "300         Prochlorococcus marinus  strain=CCMP1375; SS120   \n",
       "301     Prochlorococcus_A marinus_S             strain=MED4   \n",
       "302     Prochlorococcus_C marinus_B          strain=MIT9313   \n",
       "303     Prochlorococcus_B marinus_B           strain=NATL2A   \n",
       "304     Prochlorococcus_A marinus_L         strain=MIT 9312   \n",
       "...                             ...                     ...   \n",
       "1130    Prochlorococcus_B marinus_B                     NaN   \n",
       "1131  Prochlorococcus_A sp003211315                     NaN   \n",
       "1132    Prochlorococcus_B marinus_B                     NaN   \n",
       "1133  Prochlorococcus_A sp003210915                     NaN   \n",
       "1134    Prochlorococcus_A marinus_S                     NaN   \n",
       "\n",
       "                                                  label  \\\n",
       "300   c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...   \n",
       "301   c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...   \n",
       "302   c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...   \n",
       "303   c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...   \n",
       "304   c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...   \n",
       "...                                                 ...   \n",
       "1130  c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...   \n",
       "1131  c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...   \n",
       "1132  c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...   \n",
       "1133  c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...   \n",
       "1134  c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...   \n",
       "\n",
       "                                       protein_filename  \\\n",
       "300   GCA_000007925.1_ASM792v1_genomic_prodigal_prot...   \n",
       "301   GCA_000011465.1_ASM1146v1_genomic_prodigal_pro...   \n",
       "302   GCA_000011485.1_ASM1148v1_genomic_prodigal_pro...   \n",
       "303   GCA_000012465.1_ASM1246v1_genomic_prodigal_pro...   \n",
       "304   GCA_000012645.1_ASM1264v1_genomic_prodigal_pro...   \n",
       "...                                                 ...   \n",
       "1130  GCA_003211895.1_ASM321189v1_genomic_prodigal_p...   \n",
       "1131  GCA_003211315.1_ASM321131v1_genomic_prodigal_p...   \n",
       "1132  GCA_003211155.1_ASM321115v1_genomic_prodigal_p...   \n",
       "1133  GCA_003210915.1_ASM321091v1_genomic_prodigal_p...   \n",
       "1134  GCA_003208845.1_ASM320884v1_genomic_prodigal_p...   \n",
       "\n",
       "                                              gtt_label  \n",
       "300   c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...  \n",
       "301   c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...  \n",
       "302   c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...  \n",
       "303   c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...  \n",
       "304   c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...  \n",
       "...                                                 ...  \n",
       "1130  c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...  \n",
       "1131  c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...  \n",
       "1132  c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...  \n",
       "1133  c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...  \n",
       "1134  c_Cyanobacteriia_o_Synechococcales_f_Cyanobiac...  \n",
       "\n",
       "[483 rows x 63 columns]"
      ]
     },
     "execution_count": 859,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[~df['is_synecho'] & (df['gtdb_order'] == 'Synechococcales') & (df[\"gtdb_genus\"] != \"Prochlorothrix\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 864,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>biosample</th>\n",
       "      <th>wgs_master</th>\n",
       "      <th>excluded_from_refseq</th>\n",
       "      <th>refseq_category</th>\n",
       "      <th>relation_to_type_material</th>\n",
       "      <th>taxid</th>\n",
       "      <th>species_taxid</th>\n",
       "      <th>isolate</th>\n",
       "      <th>...</th>\n",
       "      <th>gtdb_species</th>\n",
       "      <th>gc</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>infraspecific_name</th>\n",
       "      <th>new_genus</th>\n",
       "      <th>new_species</th>\n",
       "      <th>new_strain</th>\n",
       "      <th>label</th>\n",
       "      <th>protein_filename</th>\n",
       "      <th>gtt_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>97</td>\n",
       "      <td>GCA_000775285.1</td>\n",
       "      <td>PRJNA244289</td>\n",
       "      <td>SAMN02725026</td>\n",
       "      <td>JJML00000000.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>representative genome</td>\n",
       "      <td>assembly from type material</td>\n",
       "      <td>1497020</td>\n",
       "      <td>1501145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Neosynechococcus sphagnicola</td>\n",
       "      <td>0.51590</td>\n",
       "      <td>Neosynechococcus sphagnicola sy1</td>\n",
       "      <td>strain=CAUP A 1101</td>\n",
       "      <td>Neosynechococcus</td>\n",
       "      <td>sphagnicola</td>\n",
       "      <td>CAUP_A_1101</td>\n",
       "      <td>Neosynechococcus_sphagnicola_CAUP_A_1101</td>\n",
       "      <td>GCA_000775285.1_ASM77528v1_genomic_prodigal_pr...</td>\n",
       "      <td>Neosynechococcus_sphagnicola_CAUP_A_1101 # GCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>734</td>\n",
       "      <td>GCA_001939115.1</td>\n",
       "      <td>PRJNA355315</td>\n",
       "      <td>SAMN06074267</td>\n",
       "      <td>MQTZ00000000.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>na</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1922337</td>\n",
       "      <td>1922337</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>GCF-001939115 sp001939115</td>\n",
       "      <td>0.52317</td>\n",
       "      <td>Leptolyngbya sp. 'hensonii'</td>\n",
       "      <td>strain=hensonii</td>\n",
       "      <td>GCF-001939115</td>\n",
       "      <td>GCF-001939115 sp001939115</td>\n",
       "      <td>strain=hensonii</td>\n",
       "      <td>c_Cyanobacteriia_o_Neosynechococcales_f_Neosyn...</td>\n",
       "      <td>GCA_001939115.1_ASM193911v1_genomic_prodigal_p...</td>\n",
       "      <td>c_Cyanobacteriia_o_Neosynechococcales_f_Neosyn...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 63 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    assembly_accession   bioproject     biosample      wgs_master  \\\n",
       "97     GCA_000775285.1  PRJNA244289  SAMN02725026  JJML00000000.1   \n",
       "734    GCA_001939115.1  PRJNA355315  SAMN06074267  MQTZ00000000.1   \n",
       "\n",
       "    excluded_from_refseq        refseq_category    relation_to_type_material  \\\n",
       "97                   NaN  representative genome  assembly from type material   \n",
       "734                  NaN                     na                          NaN   \n",
       "\n",
       "       taxid  species_taxid isolate  ...                  gtdb_species  \\\n",
       "97   1497020        1501145     NaN  ...  Neosynechococcus sphagnicola   \n",
       "734  1922337        1922337     NaN  ...     GCF-001939115 sp001939115   \n",
       "\n",
       "          gc                     organism_name  infraspecific_name  \\\n",
       "97   0.51590  Neosynechococcus sphagnicola sy1  strain=CAUP A 1101   \n",
       "734  0.52317       Leptolyngbya sp. 'hensonii'     strain=hensonii   \n",
       "\n",
       "            new_genus                new_species       new_strain  \\\n",
       "97   Neosynechococcus                sphagnicola      CAUP_A_1101   \n",
       "734     GCF-001939115  GCF-001939115 sp001939115  strain=hensonii   \n",
       "\n",
       "                                                 label  \\\n",
       "97            Neosynechococcus_sphagnicola_CAUP_A_1101   \n",
       "734  c_Cyanobacteriia_o_Neosynechococcales_f_Neosyn...   \n",
       "\n",
       "                                      protein_filename  \\\n",
       "97   GCA_000775285.1_ASM77528v1_genomic_prodigal_pr...   \n",
       "734  GCA_001939115.1_ASM193911v1_genomic_prodigal_p...   \n",
       "\n",
       "                                             gtt_label  \n",
       "97   Neosynechococcus_sphagnicola_CAUP_A_1101 # GCA...  \n",
       "734  c_Cyanobacteriia_o_Neosynechococcales_f_Neosyn...  \n",
       "\n",
       "[2 rows x 63 columns]"
      ]
     },
     "execution_count": 864,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.query('gtdb_family == \"Neosynechococcaceae\"')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 867,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Parasynechococcus      53\n",
       "Pseudosynechococcus    41\n",
       "Synechospongium        30\n",
       "Limnothrix             14\n",
       "Inmanicoccus           12\n",
       "Leptococcus            10\n",
       "Regnicoccus             9\n",
       "Cyanobium               9\n",
       "Thermosynechococcus     6\n",
       "Synechococcus           6\n",
       "Vulcanococcus           4\n",
       "Lacustricoccus          3\n",
       "Leptovivax              1\n",
       "Brevicoccus             1\n",
       "Stenotopis              1\n",
       "Neosynechococcus        1\n",
       "Coccusdissimilis        1\n",
       "Name: new_genus, dtype: int64"
      ]
     },
     "execution_count": 867,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['is_synecho']]['new_genus'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 885,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>label</th>\n",
       "      <th>seq_rel_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>Limnothrix_euryhalinus_PCC 7002</td>\n",
       "      <td>2008/03/14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>53</td>\n",
       "      <td>Limnothrix_euryhalinus_PCC_8807</td>\n",
       "      <td>2016/07/25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>56</td>\n",
       "      <td>Limnothrix_euryhalinus_PCC_7117</td>\n",
       "      <td>2016/07/25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>72</td>\n",
       "      <td>Limnothrix_sp._PCC_7003</td>\n",
       "      <td>2016/07/25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73</td>\n",
       "      <td>Limnothrix_euryhalinus_PCC_73109</td>\n",
       "      <td>2016/01/21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>83</td>\n",
       "      <td>Limnothrix_euryhalinus_NKBG042902</td>\n",
       "      <td>2014/06/27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>96</td>\n",
       "      <td>Limnothrix_sp._NKBG15041c</td>\n",
       "      <td>2013/10/22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>104</td>\n",
       "      <td>Limnothrix_sp._NIES-970</td>\n",
       "      <td>2017/02/08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>107</td>\n",
       "      <td>Limnothrix_euryhalinus_PCC_7002</td>\n",
       "      <td>2017/05/13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>108</td>\n",
       "      <td>Limnothrix_euryhalinus_OG1</td>\n",
       "      <td>2017/04/21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>133</td>\n",
       "      <td>Limnothrix_euryhalinus_BDU_130192</td>\n",
       "      <td>2017/10/25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>211</td>\n",
       "      <td>Limnothrix_Euryhalinus_PCC_11901</td>\n",
       "      <td>2019/05/22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>378</td>\n",
       "      <td>c_Cyanobacteriia_o_Cyanobacteriales_f_Limnotri...</td>\n",
       "      <td>2016/12/15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>426</td>\n",
       "      <td>c_Cyanobacteriia_o_Cyanobacteriales_f_Limnotri...</td>\n",
       "      <td>2012/12/04</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 label seq_rel_date\n",
       "11                     Limnothrix_euryhalinus_PCC 7002   2008/03/14\n",
       "53                     Limnothrix_euryhalinus_PCC_8807   2016/07/25\n",
       "56                     Limnothrix_euryhalinus_PCC_7117   2016/07/25\n",
       "72                             Limnothrix_sp._PCC_7003   2016/07/25\n",
       "73                    Limnothrix_euryhalinus_PCC_73109   2016/01/21\n",
       "83                   Limnothrix_euryhalinus_NKBG042902   2014/06/27\n",
       "96                           Limnothrix_sp._NKBG15041c   2013/10/22\n",
       "104                            Limnothrix_sp._NIES-970   2017/02/08\n",
       "107                    Limnothrix_euryhalinus_PCC_7002   2017/05/13\n",
       "108                         Limnothrix_euryhalinus_OG1   2017/04/21\n",
       "133                  Limnothrix_euryhalinus_BDU_130192   2017/10/25\n",
       "211                   Limnothrix_Euryhalinus_PCC_11901   2019/05/22\n",
       "378  c_Cyanobacteriia_o_Cyanobacteriales_f_Limnotri...   2016/12/15\n",
       "426  c_Cyanobacteriia_o_Cyanobacteriales_f_Limnotri...   2012/12/04"
      ]
     },
     "execution_count": 885,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['gtdb_family'] == 'Limnotrichaceae'][['label', 'seq_rel_date']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create AAI inputs at family level"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 882,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(166, 63)"
      ]
     },
     "execution_count": 882,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# # Create AAI input for Synechococcus - added Cyanobium and Limnothrix\n",
    "dst = \"../data/comparem/input/\"\n",
    "\n",
    "df['proteins_path'] = df['proteins_path'].str.replace(\"..//ncbi\", \"../data/ncbi/\")\n",
    "\n",
    "subsets = dict()\n",
    "\n",
    "# Cyanobiaceae\n",
    "subsets['cyanobiaceae'] = df[(df['gtdb_family'] == 'Cyanobiaceae') & df['is_synecho'] | df['label'].apply(lambda s:\n",
    "                                                                                any(i in s for i in [\"SS35\", \n",
    "                                                                                                     \"SS2\", \n",
    "                                                                                                     \"SS52\", \n",
    "                                                                                                     \"CCMP1375\", \n",
    "                                                                                                     \"LG\",]))]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 876,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1085, 63)"
      ]
     },
     "execution_count": 876,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
